evm_arithmetization/src/generation/prover_input.rs

```diff
@@ -383,6 +451,47 @@ impl<F: Field> GenerationState<F> {
             }
         }
     }
+
+    pub(crate) fn get_addresses_access_list(&self) -> Result<AccList, ProgramError> {
```

Comment:
This seems to be called at every access list `PROVER_INPUT` call, of which there might be quite a lot for big transactions. I don't have any numbers (and I don't see an easy way to do it with the current code), but maybe it would be worth it to maintain the list as a field of `GenerationState` to reduce overhead?

Reply:
Maybe I can change `access_list_mem` field from `AccList` to be just a pointer to the corresponding memory segment in `self`